SampleApp/OnboardingViewController.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  OnboardingViewController.swift
+//  SampleApp
+//
+//  Created by Kian on 2020-10-19.
+//  Copyright © 2020 MotionTag GmbH. All rights reserved.
+//
+
+import UIKit
+
+protocol OnboardingCompleteDelegage: class {
+    func onboardingDidEnd()
+}
+
+class OnboardingViewController: UIViewController {
+
+    // Must be replaced with a valid token: https://api.motion-tag.de/developer/
+    private let userJwtToken = "User's JWT token"
+
+    @IBOutlet weak var loginButton: UIButton!
+    @IBOutlet weak var locationButton: UIButton!
+    @IBOutlet weak var activityButton: UIButton!
+    @IBOutlet weak var endOnboardingButton: UIButton!
+
+    weak var delegate: OnboardingCompleteDelegage?
+    
+    private lazy var permissions: Permissions = {
+        let manager = Permissions()
+        manager.locationDelegate = self
+        manager.activityDelegate = self
+        return manager
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Onboarding"
+        view.backgroundColor = .white
+    }
+
+    @IBAction func loginTapped(_ sender: Any) {
+        loginButton.isEnabled = false
+        loginButton.backgroundColor = .green
+        PersistenceLayer.token = userJwtToken
+    }
+
+    @IBAction func locationAuthTapped(_ sender: Any) {
+        locationButton.isEnabled = false
+        permissions.obtainLocationPermission()
+    }
+
+    @IBAction func activityAuthTapped(_ sender: Any) {
+        activityButton.isEnabled = false
+        permissions.obtainMotionActivityPermission()
+    }
+
+    @IBAction func endOnboardingTapped(_ sender: Any) {
+        PersistenceLayer.isOnboardingOver = true
+        if let delegate = delegate {
+            delegate.onboardingDidEnd()
```

Comment:
just in one line: 
delegate?.onboardingDidEnd()

Reply:
true 👍